ms2pip/predict_xgboost.py

```diff
@@ -0,0 +1,164 @@
+"""Get predictions directly from XGBoost model, within ms2pip framework."""
+
+from genericpath import isfile
+from itertools import islice
+import os
+import urllib.request
+import logging
+
+
+import numpy as np
+import xgboost as xgb
+import hashlib
+
+from ms2pip.ms2pipC import AMINO_ACID_IDS, ms2pip_pyx
+from ms2pip.exceptions import (
+    InvalidModificationFormattingError,
+    InvalidXGBoostModelError,
+    UnknownModificationError,
+    UnknownFragmentationMethodError,
+)
+logger = logging.getLogger("ms2pip")
+
+
+def process_peptides_xgb(peprec, model_params, ptm_ids, num_cpu=1):
```

Comment:
Maybe we should split `process_peptides` from ms2pipC to have more of a similar structure?

Reply:
I do not really understand what you mean :p is it necessary to do before the immunopeptide model release?

Reply:
TODO in the next PR